flink-connectors/flink-connector-base/src/main/java/org/apache/flink/connector/base/sink/writer/BufferedRequestState.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.connector.base.sink.writer;
+
+import org.apache.flink.annotation.PublicEvolving;
+
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Objects;
+
+/**
+ * Class holding internal state of buffered requests.
+ *
+ * @param <RequestEntryT> request type.
+ */
+@PublicEvolving
+public class BufferedRequestState<RequestEntryT extends Serializable> implements Serializable {
+    private final Collection<RequestEntryWrapper<RequestEntryT>> bufferedRequestEntries;
+    private final Long stateSize;
+
+    public BufferedRequestState(
+            Collection<RequestEntryWrapper<RequestEntryT>> bufferedRequestEntries) {
+        this.bufferedRequestEntries = bufferedRequestEntries;
+        this.stateSize =
+                bufferedRequestEntries.stream()
+                        .map(RequestEntryWrapper::getSize)
+                        .reduce(Long::sum)
+                        .orElse(0L);
+    }
+
+    public Collection<RequestEntryWrapper<RequestEntryT>> getBufferedRequestEntries() {
+        return bufferedRequestEntries;
+    }
+
+    public Long getStateSize() {
+        return stateSize;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        if (other == null || other.getClass() != this.getClass()) {
+            return false;
+        }
+        BufferedRequestState<RequestEntryT> that = (BufferedRequestState<RequestEntryT>) other;
+        return Arrays.equals(
+                this.bufferedRequestEntries.toArray(), that.bufferedRequestEntries.toArray());
+    }
+
+    @Override
+    public int hashCode() {
+        int hash = 0;
+        for (RequestEntryWrapper<RequestEntryT> request : bufferedRequestEntries) {
+            hash = Objects.hash(hash, request);
+        }
+
+        return hash;
+    }
```

Comment:
Why did you implement this? Flink coding standard advise against this unless really necessary https://flink.apache.org/contributing/code-style-and-quality-java.html